UNITTESTS/fakes/events/EventQueue.h

```diff
@@ -0,0 +1,90 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2020 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef EVENTQUEUE_FAKE_H
+#define EVENTQUEUE_FAKE_H
+
+#include <map>
+#include <memory>
+#include <functional>
+#include <algorithm>
+#include "events/EventQueue.h"
+#include <chrono>
+
+namespace events {
+
+typedef intptr_t handle_t;
+typedef std::function<void()> function_t;
+typedef unsigned long tick_t;
+
+class EventQueue {
+    using duration = std::chrono::duration<int, std::milli>;
+
+public:
+    EventQueue(unsigned size = 0, unsigned char *buffer = NULL) { delete buffer; };
+
+    ~EventQueue() { };
+
+    void dispatch(int ms = -1) {
```

Comment:
Could you add a comment explaining how the simulated dispatch works ?

Reply:
added